lib/WUI/nhttp/headers.cpp

```diff
@@ -139,4 +147,33 @@ ContentType guess_content_by_ext(const char *filename) {
     return ContentType::ApplicationOctetStream;
 }
 
+uint32_t compute_etag(const struct stat &stat) {
+    /*
+     * We need something that hopefully changes whenever someone replaces a
+     * file with a different file but the same name. Hashing a bit of metadata
+     * seems like a good candidate.
+     */
+    mbedtls_sha256_context ctx;
+    mbedtls_sha256_init(&ctx);
+    mbedtls_sha256_starts_ret(&ctx, false);
+    mbedtls_sha256_update_ret(&ctx, (const uint8_t *)&stat.st_ino, sizeof stat.st_ino);
+    mbedtls_sha256_update_ret(&ctx, (const uint8_t *)&stat.st_size, sizeof stat.st_size);
+    mbedtls_sha256_update_ret(&ctx, (const uint8_t *)&stat.st_mtime, sizeof stat.st_mtime);
```

Comment:
... also, for G-code files ordering in the filebrowser we have to use both creation time and modification time (which :window: s sets separately unlike linux), so I'd better use both times for computing hash.

Reply:
Good tip. It's not worth picking the newest, but taking both is better ‒ better chance a different file won't provide the same values.